src/main/java/backend/IssueMetadata.java

```diff
@@ -49,17 +65,27 @@ public IssueMetadata(List<TurboIssueEvent> events, List<Comment> comments) {
         this.events = events;
         this.comments = comments;
         this.nonSelfUpdatedAt = LocalDateTime.ofEpochSecond(0, 0, ZoneOffset.ofHours(0)); // Not calculated yet
+        this.selfUpdatedAt = LocalDateTime.ofEpochSecond(0, 0, ZoneOffset.ofHours(0)); // Not calculated yet
         this.nonSelfCommentCount = 0; // Not calculated yet
+        this.selfCommentCount = 0; // Not calculated yet
         this.isUpdated = false;
+        this.isUpdatedByOthers = false;
+        this.isUpdatedBySelf = false;
     }
 
     // Constructor used in Logic
-    public IssueMetadata(IssueMetadata other, LocalDateTime nonSelfUpdatedAt, int nonSelfCommentCount) {
+    public IssueMetadata(IssueMetadata other, LocalDateTime nonSelfUpdatedAt, LocalDateTime selfUpdatedAt,
+                         int nonSelfCommentCount, int selfCommentCount,
+                         boolean isUpdatedByOthers, boolean isUpdatedBySelf) {
         this.events = new ArrayList<>(other.events);
         this.comments = new ArrayList<>(other.comments);
         this.nonSelfUpdatedAt = nonSelfUpdatedAt; // Calculated just prior to calling this constructor
-        this.nonSelfCommentCount = nonSelfCommentCount; // Calculated just prior to calling
+        this.selfUpdatedAt = selfUpdatedAt;
+        this.nonSelfCommentCount = nonSelfCommentCount; // Calculated just prior to calling this constructor
+        this.selfCommentCount = selfCommentCount; //Calculated just prior to calling this constructor
         this.isUpdated = true;
+        this.isUpdatedByOthers = isUpdatedByOthers; //Calculated prior to calling this constructor
+        this.isUpdatedBySelf = isUpdatedBySelf; // Calculated prior to calling this constructor
```

Comment:
I don't think these comments are necessary. The constructor receives these values as parameters so it's kind of a given.